File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    clo = {}
    for i in range(len(clothes)):
        clo[clothes[i][1]]=0
    for i in range(len(clothes)):
        clo[clothes[i][1]]+=1
    c = list(clo.values())
    ans = 1
    for i in range(len(c)):
        c[i]+=1
        ans *= c[i]
    return ans-1
    

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let map = new Map()
    for(let c of clothes){
        if(map.has(c[1])) map.set(c[1],map.get(c[1])+1)
        else map.set(c[1],1)
    }
    let arr = [...map.values()]
    let ans = 1
    for(let i=0;i<arr.length;i++){
        ans *= (arr[i]+1)
    }
    return ans-1
}